lib/proxy.js

```diff
@@ -68,21 +68,63 @@ function getProxyUri(uri) {
 }
 
 
+/**
+ * @typedef {Object} UrlAndBasicAuth
+ * @property {string} url - url without basic auth
+ * @property {string} basicAuth auth part of url
+ */
+
+/**
+ * Separates baic auth credentials from the originUrl.
+ *
+ * Many services support basic authentication as part of URLs. 
+ * For example http://me:secret@myhost.com has credentials for user me with password secret.
+ * The here used module #http-proxy however, doesn't supports this URL. Instead it requires
+ * an extra option for basic auth.
+ *
+ * This method separates the basic auth from the originUrl.
+ *
+ * @param {string} originUrl original URL, which can contain basic auth
+ * @returns {UrlAndBasicAuth}
+ * }
+ */
+function separateBasicAuth(originUrl){
```

Comment:
I wonder why it's necessary to manually parse the url as `url.parse` should do so (`auth` property).